services/contract_call.proto

```diff
@@ -59,4 +59,9 @@ message ContractCallTransactionBody {
      * which function to call, and the parameters to pass to the function
      */
     bytes functionParameters = 4;
+
+    /**
+     * List with accounts or smart contracts and storage keys that should be "warmed up"/loaded in the EVM
+     */
+    repeated AccessListEntry accessList = 5;
```

Comment:
One possibility is instead of having an AccessListEntry that maps directly to the Ethereum model if we have two fields, `repeated AccountId accountAccesses = 5;` and `repeated ContractAccecssEntry contractAccesses = 6;` and the `AccessListEntry` only has `ContractID` and the repeated storageKeys entry.  This would also tidy up the logic in the implementation with less if clauses.

Reply:
I was thinking about the same problem and wondered whether if its an overkill to add 2 message types, but since you are suggesting the same, I think that we should do it that way and add separate messages for accounts and contracts 👍